sqlmesh/core/context.py

```diff
@@ -1867,10 +1884,13 @@ def print_info(self, skip_connection: bool = False, verbose: bool = False) -> No
                 self.config.get_state_connection(self.gateway), self.console, "State Connection"
             )
 
-        self._try_connection("data warehouse", self._engine_adapter.ping)
+        self._create_engine_adapters()
+        self._try_connection(
+            "data warehouse", [adapter.ping for adapter in self._engine_adapters.values()]
```

Comment:
On a second thought, I don't think this is valid. Eg. the multiple gateways may have different state and test connections but we don't test those. I'd say when using `sqlmesh info` users are expected to specify which specific gateway they want info for, like `sqlmesh --gateway <target gateway> info`.

Reply:
Additionally, if we remove this logic from here, the `_create_engine_adapters` method will only be called from `def snapshot_evaluator`, which means that we no longer need to mutate the `_engine_adapters` state within the context and just return the dict from the method itself.

Reply:
If we do want to keep this logic, I'd suggest extracting the actual gateways from the models and only testing those.

Reply:
> On a second thought, I don't think this is valid. Eg. the multiple gateways may have different state and test connections but we don't test those. I'd say when using sqlmesh info users are expected to specify which specific gateway they want info for, like sqlmesh --gateway <target gateway> info.

Right this use is actually more clear to test the specified engine when using info with a specific gateway, so removing this logic to revert to how it was originally

Reply:
Ah, nice. I think this was a misleading suggestion on my end, the `--gateway` idea is indeed much better 👍